tests/app/v2/notifications/test_post_notifications.py

```diff
@@ -651,9 +651,9 @@ def test_post_email_notification_validates_personalisation_send_a_file_values(
     expected_filename,
 ):
     mocker.patch("app.celery.provider_tasks.deliver_email.apply_async")
-    document_download_mock = mocker.patch("app.v2.notifications.post_notifications.document_download_client")
-    document_download_mock.upload_document.side_effect = (
-        lambda service_id, content, is_csv, confirmation_email, **kwargs: f"{content}-link"
+    document_download_upload_document_mock = mocker.patch(
+        "app.document_download_client.upload_document",
```

Comment:
Bit of a nitpick, but couldn't you have achieved the same effect with a smaller/easier to read diff by just changing `"app.v2.notifications.post_notifications.document_download_client"` to `"app.document_download_client"` (leaving the rest as-is)?

Reply:
It wouldn't be _exactly_ the same effect, no.

Patching the `document_download_client` will cause the name binding _in the target module_ to be switched out to a mock object. Any code that accesses the `document_download_client` from somewhere else will get the original unmocked client.

Patching `document_download_client.upload_document` will patch the referenced object's method, which should have an effect globally.

When I see the former it gives me anxiety, when I see the latter I have more confidence it's been mocked adequately.